modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLMetadataTestCase.java

```diff
@@ -0,0 +1,110 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.integration.test.saml;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.tenant.mgt.stub.TenantMgtAdminServiceExceptionException;
+import org.wso2.identity.integration.common.clients.TenantManagementServiceClient;
+import org.wso2.identity.integration.common.utils.ISIntegrationTest;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.json.JSONObject;
+import org.json.XML;
+
+import java.io.IOException;
+
+public class SAMLMetadataTestCase extends ISIntegrationTest {
+
+    private static final String SUPER_TENANT_URL = "https://localhost:9853/identity/metadata/saml2";
+    private static final String TENANT_URL = "https://localhost:9853/t/wso2.com/identity/metadata/saml2";
+    private static final String SAML_METADATA_ENDPOINT_TENANT = "https://localhost:9853/samlsso?tenantDomain=wso2.com";
+    private static final String SAML_METADATA_ENDPOINT_SUPER_TENANT = "https://localhost:9853/samlsso";
+    private static final String SAMLARTRESOLVE_ENDPOINT = "https://localhost:9853/samlartresolve";
+    private TenantManagementServiceClient tenantServiceClient;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        tenantServiceClient = new TenantManagementServiceClient(isServer.getContextUrls().getBackEndUrl(),
+                sessionCookie);
+    }
+
+    @Test(groups = "wso2.is", description = "This test method will test SAML Metadata endpoints.")
+    public void getSAMLMetadata() throws IOException, TenantMgtAdminServiceExceptionException, JSONException {
+
+        testResponseContent(SUPER_TENANT_URL, SAML_METADATA_ENDPOINT_SUPER_TENANT);
+        tenantServiceClient.addTenant("wso2.com", "John", "password",
+                "john@friends.com", "John", "Smith");
+        testResponseContent(TENANT_URL, SAML_METADATA_ENDPOINT_TENANT);
+    }
+
+    private void testResponseContent(String locationURL, String samlMetadataEndpoint)
+            throws IOException, JSONException {
+
+        HttpClient client = HttpClientBuilder.create().build();
+        HttpResponse httpResponse = sendGetRequest(client, locationURL);
+        String content = DataExtractUtil.getContentData(httpResponse);
+        Assert.assertNotNull(content);
+        JSONArray singleLogoutServices = XML.toJSONObject(content).getJSONObject("EntityDescriptor").getJSONObject(
+                "IDPSSODescriptor").getJSONArray("SingleLogoutService");
+
+        for (int i = 0; i < singleLogoutServices.length(); i++) {
+
```

Comment:
At the beginning of the for loop a new line is not required. Shall we remove this? WDYT?

Reply:
Fixed https://github.com/wso2/product-is/pull/8319/commits/ed4daeda51db12ad05cd8d718640a10f984fe877